pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -927,6 +927,8 @@ private void createNewSegmentZKMetadata(TableConfig tableConfig, StreamConfig st
       newSegmentZKMetadata.setPartitionMetadata(partitionMetadata);
     }
 
+    newSegmentZKMetadata.setPreviousSegment(committingSegmentDescriptor.getSegmentName());
```

Comment:
Could this be `null`? We should probably check if `committingSegmentZKMetadata == null`